vllm/v1/engine/core_client.py

```diff
@@ -215,15 +207,21 @@ def __init__(self, vllm_config: VllmConfig,
 
     def get_output(self) -> EngineCoreOutputs:
 
-        (frame, ) = self.output_socket.recv_multipart(copy=False)
-        return self.decoder.decode(frame.buffer)
+        try:
+            (frame, ) = self.output_socket.recv_multipart(copy=False)
+            self._validate_alive(frame.buffer)
+            return self.decoder.decode(frame.buffer)
+        except Exception as e:
+            raise self._format_exception(e) from None
```

Comment:
You called this out in the PR description - I think it'd be worth filling an issue to help us remember to come back and look at zmq error handling. Depending on the error, our handling should differ. For example, a `ZMQError` exception with an `errno` of `EAGAIN` means we should try again as opposed to reporting an error.